rfqmath/fixed_point.go

```diff
@@ -103,6 +103,53 @@ func (f FixedPoint[T]) Equals(other FixedPoint[T]) bool {
 	return f.Coefficient.Equals(other.Coefficient) && f.Scale == other.Scale
 }
 
+// WithinTolerance returns true if the two FixedPoint values are within the
+// given tolerance (in parts per million (PPM)).
+func (f FixedPoint[T]) WithinTolerance(
+	other FixedPoint[T], tolerancePpm T) bool {
+
+	// Determine the larger scale between the two fixed-point numbers.
+	// Both values will be scaled to this larger scale to ensure a
+	// consistent comparison.
+	var largerScale uint8
+	if f.Scale > other.Scale {
+		largerScale = f.Scale
+	} else {
+		largerScale = other.Scale
+	}
```

Comment:
Ignorant comment: Would the case where, if even possible, the FixedPoints are equal, need to be handled? or adding a general error if `largerScale` isn't set

Reply:
I've added a test case where both fixed-points are exactly equal:
```
		{
			// Case where firstFp and secondFp are equal,
			// tolerance is 0 PPM. Diff within bounds.
			firstFp:      FixedPointFromUint64[N](100_000, 2),
			secondFp:     FixedPointFromUint64[N](100_000, 2),
			tolerancePpm: 0, // 0% tolerance in PPM
			withinBounds: true,
		},
```